File: JavaScript/8js 3/twoSort.js
/*
|--------------------------------------------------------------------------
| Sort and Star
|--------------------------------------------------------------------------
|
*/

function twoSort(s) {
   return s.sort()[0].split('').join('***');
}


const twoSort = s => [...s.sort()[0]].join('***');


function twoSort(s) {
   let last = s.sort()[0];
   let a = last[0];
   for (let i = 1; i < last.length; i++) {
      a += "***" + last[i];
   }
   return a;
}


function twoSort(array) {
   return [...array.reduce((min, str) => min < str ? min : str)].join("***");
}


function twoSort(q) {
   let a = q.sort();
   a = a[0].replace(/(.)(?=.{1,}$)/g, '$1***');
   return a;
}


console.log(twoSort(["bitcoin", "take", "over", "the", "world", "maybe", "who", "knows", "perhaps"]), 'b***i***t***c***o***i***n');
console.log(twoSort(["turns", "out", "random", "test", "cases", "are", "easier", "than", "writing", "out", "basic", "ones"]), 'a***r***e');

File: Python/8py 3/two_sort.py
"""
|--------------------------------------------------------------------------
| Sort and Star
|--------------------------------------------------------------------------
|
"""

def two_sort(array):
   return '***'.join(list(sorted(array)[0]))


def two_sort(a):
   a = sorted(a)
   result = a[0]
   result = result.replace("", "***")
   return result [3:-3]


def two_sort(array):
   return '***'.join(min(array))


def two_sort(array):
   array.sort()
   q = []
   for i in array[0]:
      q.append(i + '***')
   return ''.join(q).rstrip('***')


print(two_sort(["bitcoin", "take", "over", "the", "world", "maybe", "who", "knows", "perhaps"]), 'b***i***t***c***o***i***n' )
print(two_sort(["turns", "out", "random", "test", "cases", "are", "easier", "than", "writing", "out", "basic", "ones"]), 'a***r***e')
print(two_sort(["lets", "talk", "about", "javascript", "the", "best", "language"]), 'a***b***o***u***t')
print(two_sort(["i", "want", "to", "travel", "the", "world", "writing", "code", "one", "day"]), 'c***o***d***e')
print(two_sort(["Lets", "all", "go", "on", "holiday", "somewhere", "very", "cold"]), 'L***e***t***s')